hoomd/conftest.py

```diff
@@ -271,3 +324,128 @@ def operation_pickling_check(instance, sim):
     sim.operations += instance
     sim.run(0)
     pickling_check(instance)
+
+
+class BlockAverage:
+    """Block average method for estimating standard deviation of the mean.
+
+    Args:
+        data: List of values
+    """
+
+    def __init__(self, data):
+        # round down to the nearest power of 2
+        N = 2**int(math.floor(math.log(len(data)) / math.log(2)))
+        if N != len(data):
+            warnings.warn(
+                "Ignoring some data. Data array should be a power of 2.")
+
+        block_sizes = []
+        block_mean = []
+        block_variance = []
+
+        # take means and averages of blocks, growing blocks by factors of 2
+        block_size = 1
+        while block_size <= N // 8:
+            num_blocks = N // block_size
+            block_data = numpy.zeros(num_blocks)
+
+            for i in range(0, num_blocks):
+                start = i * block_size
+                end = start + block_size
+                block_data[i] = numpy.mean(data[start:end])
+
+            block_mean.append(numpy.mean(block_data))
+            block_variance.append(numpy.var(block_data) / (num_blocks - 1))
+
+            block_sizes.append(block_size)
+            block_size *= 2
+
+        self._block_mean = numpy.array(block_mean)
+        self._block_variance = numpy.array(block_variance)
+        self._block_sizes = numpy.array(block_sizes)
+        self.data = numpy.array(data)
+
+        # check for a plateau in the relative error before the last data point
+        block_relative_error = numpy.sqrt(self._block_variance) / numpy.fabs(
+            self._block_mean)
+        relative_error_derivative = (
+            block_relative_error[1:] - block_relative_error[:-1]) / (
+                self._block_sizes[1:] - self._block_sizes[:-1])
```

Comment:
I don't want the central difference here - I specifically want to check if the error is growing from one block size to the next.

Reply:
The name through me off. This will get you the forward difference

```suggestion
        relative_error_derivative = (
            np.diff(block_relative_error) / np.diff(self._block_sizes))
```
a bit cleaner.